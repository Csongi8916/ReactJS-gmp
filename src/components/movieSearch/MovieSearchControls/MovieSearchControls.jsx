import React, { useState } from 'react';
import NxrButton from '../../UI/NxrButton/NxrButton';
import NxrToogleButton from '../../UI/NxrToogleButton/NxrToogleButton';

import './MovieSearchControls.scss';

export default function MovieSearchControls() {
  const [searchInput, setSearchInput] = useState('');

  return (
    <div className="MovSearchControl">
      <div>
        <p className="MovSearchControl_SearchTitle">Find your movie</p>
        <input className="MovSearchControl_Input" type="text" onChange={(e) => setSearchInput(e.target.value)} />
        <NxrButton className="MovSearchControl_SearchBtn">Search</NxrButton>
      </div>
      <div className="MovSearchControl_SearchMode">
        <NxrToogleButton title="Search by" firstTitle="title" secondTitle="gengre" />
      </div>
    </div>
  )
}
