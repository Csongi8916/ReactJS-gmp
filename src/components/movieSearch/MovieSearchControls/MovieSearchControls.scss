$nrWhiteColor: #ffffff;

.MovSearchControl_SearchTitle {
  text-transform: uppercase;
  font-size: 38px;
  color: $nrWhiteColor;
}

.MovSearchControl_Input {
  background-color: #424242;
  color: #fff;
  outline: none;
  border: none;
  font-size: 20px;
  padding: 10px 10px;
  line-height: 28px; 
}