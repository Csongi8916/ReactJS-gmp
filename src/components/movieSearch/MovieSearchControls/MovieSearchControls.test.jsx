import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import MovieSearchControls from './MovieSearchControls';

describe('MovieSearchControls Snapshpt tests:', () => {
  test('Should render correctly MovieCard', () => {
    const tree = renderer.create(<MovieSearchControls />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MovieSearchControls UI tests:', () => {
  test('Should change search input', () => {
    const wrap = shallow(<MovieSearchControls />);
    const input = wrap.find('.MovSearchControl_Input');
  });
});