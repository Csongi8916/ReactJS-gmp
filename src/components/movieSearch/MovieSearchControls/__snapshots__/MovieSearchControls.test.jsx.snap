// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchControls Snapshpt tests: Should render correctly MovieCard 1`] = `
<div
  className="MovSearchControl"
>
  <div>
    <p
      className="MovSearchControl_SearchTitle"
    >
      Find your movie
    </p>
    <input
      className="MovSearchControl_Input"
      onChange={[Function]}
      type="text"
    />
    <button
      className="NrxBtn"
    >
      Search
    </button>
  </div>
  <div
    className="MovSearchControl_SearchMode"
  >
    <div
      className="NxrToogle"
    >
      <span
        className="NxrToogle_ModeTitle"
      >
        Search by
      </span>
      <div
        className="NxrToogle_Container"
      >
        <button
          className="NxrToogle_Btn NxrToogle_Btn-Active"
          onClick={[Function]}
        >
          title
        </button>
        <button
          className="NxrToogle_Btn NxrToogle_Btn-Inactive"
          onClick={[Function]}
        >
          gengre
        </button>
      </div>
    </div>
  </div>
</div>
`;
